ring-core/src/ring/middleware/file.clj

```diff
@@ -40,14 +40,20 @@
   Accepts the following options:
 
   :index-files?    - look for index.* files in directories, defaults to true
-  :allow-symlinks? - serve files through symbolic links, defaults to false"
+  :allow-symlinks? - serve files through symbolic links, defaults to false
+  :handler-first? - let handler process the files first. if handler successfully processes the file, do not serve static content"
   ([handler root-path]
    (wrap-file handler root-path {}))
   ([handler root-path options]
    (ensure-dir root-path)
    (fn
      ([request]
-      (or (file-request request root-path options) (handler request)))
+      (if (:handler-first? options)
+        (let [{:keys [status] :as response-map} (handler request)]
```

Comment:
Prefer the name `response` instead of `response-map` for consistency. I don't think there's a need to destructure the status here when `(:status response)` could be used in the `if` statement below.